### ISSUE TITLE
Keras crossentropy with logits as loss function

### ISSUE THREAD

**Issue by ilango100** [tensorflow/tensorflow#24518]
**System information**
- TensorFlow version (you are using): 1.13rc0 (Windows GPU build)
- Are you willing to contribute it (Yes/No): **Yes**

**Current behavior/state.**
In the keras api for tensorflow there does not seem to be a function for crossentropy with logits, (without sigmoid or softmax activations). This is quite confusing for those coming from the estimators (like me). I see that a `from_logits` argument is being added in the current release, but it needs to be used with a proxy function or a lambda.

```
...
y_pred = keras.layers.Dense(10)(x)

model = keras.Model(inputs=x, outputs=y_pred)
model.compile("adam", lambda x, y: keras.losses.sparse_categorical_crossentropy(x, y, from_logits=True))
```

**Will this change the current api? How?**
Instead of introducing argument, I think it's better to have a new loss function like 
```
model.compile("adam", keras.losses.sparse_categorical_crossentropy_with_logits)
```
**or**
loss name as string:
```
model.compile("adam", "sparse_categorical_crossentropy_with_logits")
```

**Who will benefit with this feature?**
- Using logits to calculate loss is actually efficient than activation then loss, so tf should encourage using crossentropy with logits, as it did in estimator API.
- Those who switch from estimators to Keras API won't have any confusion.

**Comment by jvishnuvardhan** [comment]
@fchollet, 
please take a look at this feature request in keras to add crossentropy with logits as loss function

**Comment by ilango100** [comment]
I recently went through the code for crossentropy loss functions in r2.0 branch.

- It seems that even when we apply softmax or sigmoid activation and then loss, the activation is undone, ([see here](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/backend.py#L3926)) to recover logits for loss calculation. This answers the "efficiency" part of my question.
- When coming from Estimator API, if we need to use logits directly, we can use losses class API, such as [CategoricalCrossentropy](https://www.tensorflow.org/versions/r2.0/api_docs/python/tf/losses/CategoricalCrossentropy), which takes from_logits as parameter, so need for proxy function is eliminated:
```
...
y_pred = keras.layers.Dense(10)(x)

model = keras.Model(inputs=x, outputs=y_pred)
model.compile("adam", keras.losses.SparseCategoricalCrossentropy(from_logits=True))
```
which is neater than above lambda solution.

Thus, with the current loss implementations and API, there is no need for this issue. So, I am closing the issue myself.
